### PR TITLE
Bump to support 2.4.5

### PR DESCRIPTION
**Pull request by therealbstern**
I added the new methods in libshout-2.4.5 and deprecated the ones they deprecated.  I also added some documentation.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -271,36 +362,18 @@ impl ShoutConnBuilder {
...
-                Some(ShoutFormat::MP3) => {
-                    shout_conn_err!(sys::shout_set_format(shout, 1));
-                }
-                Some(ShoutFormat::Webm) => {
-                    shout_conn_err!(sys::shout_set_format(shout, 2));
-                }
-                Some(ShoutFormat::WebmAudio) => {
-                    shout_conn_err!(sys::shout_set_format(shout, 3));
+            if let Some(format) = self.format {
+                //shout_conn_err!(sys::shout_set_format(shout, format as u32));
```
Comment (Luminarys) [inline src/lib.rs]: Should this be removed?

**Comment by Luminarys** [comment]
Thanks for making these changes!

File: src/lib.rs (hunk 1 above)
Comment (therealbstern) [reply src/lib.rs]: Yes, I'll correct this.

Comment (therealbstern) [reply src/lib.rs]: Updated, thanks for catching that.

**Pull request merged**